dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -51,45 +56,101 @@ public Relationship create(Context context) throws SQLException, AuthorizeExcept
+                authorizeService.authorizeActionBoolean(context, relationship.getRightItem(), Constants.WRITE)) {
+                updatePlaceInRelationship(context, relationship, true);
+                return relationshipDAO.create(context, relationship);
+            } else {
                 throw new AuthorizeException(
-                    "Only administrators can modify relationship");
+                    "You do not have write rights on this relationship's items");
             }
-            updatePlaceInRelationship(context, relationship);
 
-            return relationshipDAO.create(context, relationship);
         } else {
             throw new IllegalArgumentException("The relationship given was not valid");
         }
     }
 
-    private void updatePlaceInRelationship(Context context, Relationship relationship) throws SQLException {
+    public void updatePlaceInRelationship(Context context, Relationship relationship, boolean isCreation)
+        throws SQLException, AuthorizeException {
+        Item leftItem = relationship.getLeftItem();
         List<Relationship> leftRelationships = findByItemAndRelationshipType(context,
-                                                                             relationship.getLeftItem(),
+                                                                             leftItem,
                                                                              relationship.getRelationshipType(), true);
+        Item rightItem = relationship.getRightItem();
         List<Relationship> rightRelationships = findByItemAndRelationshipType(context,
-                                                                              relationship.getRightItem(),
+                                                                              rightItem,
                                                                               relationship.getRelationshipType(),
                                                                               false);
 
-        if (!leftRelationships.isEmpty()) {
-            leftRelationships.sort((o1, o2) -> o2.getLeftPlace() - o1.getLeftPlace());
-            for (int i = 0; i < leftRelationships.size(); i++) {
-                leftRelationships.get(leftRelationships.size() - 1 - i).setLeftPlace(i + 1);
+        context.turnOffAuthorisationSystem();
+        if (!virtualMetadataPopulator.isUseForPlaceTrueForRelationshipType(relationship.getRelationshipType(), true)) {
+            if (!leftRelationships.isEmpty()) {
+                leftRelationships.sort((o1, o2) -> o2.getLeftPlace() - o1.getLeftPlace());
+                for (int i = 0; i < leftRelationships.size(); i++) {
+                    leftRelationships.get(leftRelationships.size() - 1 - i).setLeftPlace(i);
+                }
+                relationship.setLeftPlace(leftRelationships.get(0).getLeftPlace() + 1);
+            } else {
+                relationship.setLeftPlace(0);
```

Comment:
So this is this?
```
relationship.setLeftPlace(leftRelationships.size());
```
I think this would be more readable then `leftRelationships.get(0).getLeftPlace() + 1` and it would even work for the case if the relationships are empty.

And is there any reason, why the sorting and reordering is done descending and from the back of the list to the front? Should not something like this also work?

```
leftRelationships.sort((o1, o2) -> o1.getLeftPlace() - o2.getLeftPlace());
for (int i = 0; i < leftRelationships.size(); i++) {
    leftRelationships.get(i).setLeftPlace(i);
}
```

Reply:
@AlexanderS thanks for the great suggestion. It does indeed result in the same output but is more readable